libs/design-system-components/src/components/AppCreationForm/index.tsx

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import * as z from 'zod';
+import { Controller } from 'react-hook-form';
+import Button from '@akashaorg/design-system-core/lib/components/Button';
+import TextField from '@akashaorg/design-system-core/lib/components/TextField';
+import Stack from '@akashaorg/design-system-core/lib/components/Stack';
+import Divider from '@akashaorg/design-system-core/lib/components/Divider';
+import DropDown from '@akashaorg/design-system-core/lib/components/Dropdown';
+import { apply, tw } from '@twind/core';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { AkashaAppApplicationType } from '@akashaorg/typings/lib/sdk/graphql-types-new';
+import { ButtonType } from '../types/common.types';
+
+export enum FieldName {
+  extensionType = 'extensionType',
+  extensionID = 'extensionID',
+  extensionName = 'extensionName',
+}
+
+type AppCreationFormValues = {
+  extensionType: { id: string; type: AkashaAppApplicationType; title: string };
+  extensionID: '';
+  extensionName: '';
+};
+
+export type AppCreationFormProps = {
+  defaultValues?: AppCreationFormValues;
+  cancelButton: ButtonType;
+  createButton: {
+    label: string;
+    loading?: boolean;
+    handleClick: (data: AppCreationFormValues) => void;
+  };
+  errorMessage?: { fieldName: string; message: string };
+};
+
+const AppCreationForm: React.FC<AppCreationFormProps> = ({
+  defaultValues = {
+    extensionType: { id: '1', type: AkashaAppApplicationType.App, title: 'Application' },
+    extensionID: '',
+    extensionName: '',
+  },
+  cancelButton,
+  createButton,
+  errorMessage,
+}) => {
+  const {
+    control,
+    setError,
+    clearErrors,
+    handleSubmit,
+    formState: { errors, isDirty, isValid },
```

Comment:
`isDirty` and `isValid` aren't deterministic as explained on this issue https://akasha.jetbrains.space/p/awf/issues/1024

```
formState: { errors, dirtyFields, errors },

....
const isDirty = !!Object.keys(dirtyFields).length;
const isValid = !Object.keys(errors).length;
```

And the `onSubmit` needs to be changed as well.

You can see a sample of how this is used on `EditProfile` component.

Reply:
I tried using it the way you suggested but it incorrectly enabled the 'Create' button as soon as one field changes even though the remaining required fields are still empty/not touched. In my case, I wanted to disable the button completely till the required fields have been completed. I will test more and see why.

Reply:
Figured it out. Need to use a different condition for isFormDirty :)